packages/mui-base/src/FocusTrap/FocusTrap.tsx

```diff
@@ -212,76 +212,90 @@ function FocusTrap(props: FocusTrapProps): JSX.Element {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [open]);
 
-  React.useEffect(() => {
-    // We might render an empty child.
-    if (!open || !rootRef.current) {
-      return;
-    }
-
-    const doc = ownerDocument(rootRef.current);
-
-    const contain = (nativeEvent: FocusEvent | null) => {
-      const { current: rootElement } = rootRef;
+  const contain = React.useCallback(
```

Comment:
Can we move contain back into the effect? It doesn't need to be in a use callback anymore since we don't call it outside.

Reply:
I will include this in the follow up PR